slack_bolt/middleware/authorization/async_multi_teams_authorization.py

```diff
@@ -54,7 +53,9 @@ async def async_process(
                 req.context.set_authorize_result(auth_result)
                 token = auth_result.bot_token or auth_result.user_token
                 req.context["token"] = token
-                req.context["client"] = create_async_web_client(token)
```

Comment:
While writing tests for this pull request, I've found an issue where `context.client` does not respect the given `base_url` etc. As it was a blocker for writing the tests in this PR, I've fixed the issue too.